### ISSUE TITLE
automatic restart for linuxserver/wireguard container after wireguard-ui config save

### ISSUE THREAD

**Issue by kking124** [ngoduykhanh/wireguard-ui#70]
I've created a companion docker service to automatically restart a container running wireguard based on systemd.

```sh
docker run -d --name wg-monitor \
    --security-opt seccomp=unconfined \  
    --tmpfs /tmp \  
    --tmpfs /run \ 
    --tmpfs /run/lock \  
    -v /path/to/wireguard/config/folder:/etc/wireguard:ro 
    -v /sys/fs/cgroup:/sys/fs/cgroup:ro \ 
    -v /var/run/docker.sock:/var/run/docker.sock \ 
    -e CONTAINER=wireguard \ # change this to whatever the container you wish to restart is named
    kking124/wireguard-monitor
```

**NOTE: the config folder for `linuxserver/wireguard` MUST be mounted `:ro` or the wireguard container will be stuck constantly restarting**

You can find it on the Docker Hub at:

https://hub.docker.com/r/kking124/wireguard-monitor

**Comment by Kawabountou** [comment]
i dont't know if it's a mistake of my part, but when I use your docker-compose, the endpoint address is missing in the wg0.conf file.

**Comment by kking124** [comment]
Tl;dr - that's expected. You need to generate one from the `wireguard-ui` web interface by clicking "Apply Config" in the top right.

Is it causing an error, or is the file just missing?

Long explanation:

Because `linuxserver/wireguard` and `kking124/wireguard-monitor` both mount the config folder with `:ro` they can't write to it.  Normally, `linuxserver/wireguard` would write a `wg0.conf` at container start, but I'm guessing their init procedure performs a `touch` on that file, triggering a "change", which triggers the container restart.  

I'll look into changing the monitoring script so that it only triggers on a SHA1 change so that `linuxserver/wireguard` can write it's initial configuration file. It just didn't seem necessary since `wireguard-ui` overwrites it without reading it anyway.

**Comment by paklids** [comment]
You can see how I've tackled this problem within my terraform project https://github.com/paklids/flexiwire . Not perfect but the majority is found in the shell script https://github.com/paklids/flexiwire/blob/main/templates/wg_conf_restart.sh (run on a cron).
